dsdl_compiler/libcanard_dsdl_compiler/code_type_template.tmpl

```diff
@@ -0,0 +1,228 @@
+/*
+ * UAVCAN data structure definition for libcanard.
+ *
+ * Autogenerated, do not edit.
+ *
+ * Source file: ${t.source_file}
+ */
+
+#include "${t.header_filename}"
+#include "canard.h"
+
+<!--(macro generate_primary_body)--> #! type_name, service, max_bitlen, fields, constants, union, has_array
+
+ %if max_bitlen
+/**
+  * @brief ${type_name}Encode
+  * @param source : Pointer to source data struct
+  * @param msg_buf: Pointer to msg storage
+  * @param offset: Call with 0, bit offset to msg storage
+  * @param root_item: Call with 1, this for internal use
+  * @retval returns message length as bytes (when root_item == 1)
+  */
+uint32_t ${type_name}Encode(${type_name} *source, void *msg_buf, uint32_t offset, uint8_t root_item) {
+  %if union
+    /* Max Union Tag Value */
+    //assert(source->union_tag > ${(len(fields) - 1)});
```

Comment:
We have `CANARD_ASSERT()`, perhaps it makes sense to use it here instead of commenting out useful checks?